pkg/types/validation/installconfig.go

```diff
@@ -115,6 +115,10 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 	}
 	allErrs = append(allErrs, validateCloudCredentialsMode(c.CredentialsMode, field.NewPath("credentialsMode"), c.Platform.Name())...)
 
+	if c.Publish == types.InternalPublishingStrategy && c.Platform.AWS == nil && c.Platform.Azure == nil && c.Platform.GCP == nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("publish"), c.Publish, "Internal publish strategy does not work with non-cloud installations"))
```

Comment:
Nit: I would like to align the error message with our docs if possible. I was thinking of something like "Internal publish strategy is only supported on public clouds."